### ISSUE TITLE
Unnecessary concreate arguments ignored by compiler

### ISSUE THREAD

**Issue by objeck** [objeck/objeck-lang#18]
The following code should produce an error

```
class Program {
	function : Main(args: String[]) ~ Nil {
		f := Foo->New();
		f := f->Next()<String>;
	}
}

class Foo {
	New() {
	}

	method : public : Next() ~ Foo {
		return Nil;
	}
}
```

**Comment by objeck** [comment]
Fixed in comment 85ebd49

**Comment by objeck** [comment]
fixed in v5.7.2
